curriculum/challenges/english/blocks/lecture-understanding-object-oriented-programming-and-encapsulation/68c128cbd77e4ba9ed671937.md

```diff
@@ -0,0 +1,260 @@
+
+So why use properties for that instead of methods? It’s mostly about readability and convention. They make your code cleaner and easier to read.
+
+When you use a method, you always have to call it with parentheses. But with a property, you can access it just like a normal attribute using dot notation. That makes your code look simple even when it is doing extra work behind the scenes.
+
+For example, you might want to calculate a value or check that a new value is valid before saving it. Instead of calling a method for that, you can use an attribute-like way to do that.
+
+To create a property, you define a method and place the @property decorator above it. This tells Python to treat the method as a property.
+
+That takes us to getters. Here’s how to create one with the @property decorator:
+
+```py
+class Circle:
+    def __init__(self, radius):
+        self._radius = radius
+
+    @property
+    def radius(self): # A getter to get the radius
+        return self._radius
+  
+    @property
+    def area(self):  # A getter to calculate area
+        return 3.14 * (self._radius ** 2)
+
+my_circle = Circle(3)
+
+print(my_circle.radius) # 3
+print(my_circle.area) # 28.26
+```
+
+This example gets a radius and the area of a circle.
+
+Notice how we used `_radius` instead of radius inside the class. The underscore is a common Python convention to show that an attribute is meant to be private. In other words, it signifies that it’s for internal use and should not be accessed directly from outside the class.
+
+To make a setter to create the radius, for example, you have to define another method with the same name and use `@<property_name>.setter` above it:
+
+```py
+class Circle:
+    def __init__(self, radius):
+        self._radius = radius
+
+    @property
+    def radius(self):  # A getter to get the radius
+        return self._radius
+
+    @radius.setter
+    def radius(self, value):  # A setter to set the radius
+        if value <= 0:
+            raise ValueError('Radius must be positive')
+        self._radius = value
+
+my_circle = Circle(3)
+print('Initial radius:', my_circle.radius) # Initial radius: 3
+
+my_circle.radius = 8
+print('After modifying the radius:', my_circle.radius) # After modifying the radius: 8
+```
+
+In this example, the radius setter is not just setting the radius for the circle, it's also running a validation that makes sure the radius is not a negative number.
+
+Once you define getters and setters, Python automatically calls them under the hood whenever you use normal attribute syntax:
+
+```py
+my_circle.radius # This will call the getter
+my_circle.radius = 4 # This will call the setter
+```
+
+Note that inside the setter, you cannot use same name of the property when assigning a new value. That’s because `self.radius = value` will call the setter within the setter method itself, leading to infinite recursion and a `RecursionError`. So you must always use the underscore-prefixed form `self._radius = value`.
```

Comment:
```suggestion
Note that inside the setter, you cannot use same name of the property when assigning a new value. That's because `self.radius = value` will call the setter within the setter method itself, leading to infinite recursion and a `RecursionError`. So you must always use the underscore-prefixed form `self._radius = value`.
```